spinalcordtoolbox/scripts/sct_process_segmentation.py

```diff
@@ -363,11 +364,11 @@ def main(argv=None):
     file_out = os.path.abspath(arguments.o)
     append = bool(arguments.append)
     if arguments.vert is not None:
-        vert_levels = arguments.vert
-        fname_vert_levels = arguments.vertfile
+        levels = arguments.vert
+        vert_level = arguments.vertfile
```

Comment:
Oh gosh, [336e05e](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3860/commits/336e05e6b7a26f356b5153f6d0d923cf22cbc7e4) ended up being a bigger change than I expected. Sorry about that! :sweat_smile:

Reply:
Ok, I've added a `fname_` prefix to `vert_level` everywhere, as per [our naming convention](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Naming-conventions-(variables,-files)), and propagated the name change to all callers of the function, recursively. The name is maybe still not ideal, but still an improvement.

As for `Image()` accepting `Image` objects, I think that's fairly pythonic (for example, calling `int()` on an integer or `str()` on a string is a no-op-except-maybe-copying operation), and is a nice way of ensuring an object has the correct type. But for other functions, I think it's **not** typical to propagate this ambiguity; usually it's the caller's responsibility to ensure that arguments passed in have the right type, or at least the right duck-type. (And `str` and `Image` are not duck-type equivalent, even though you can convert between them easily, since they don't have the same methods.)

My pie-in-the-sky dream on that front would be that "API" functions in our codebase should manipulate actual Python objects, rather than, for example, strings that can be parsed/manipulated into the data that's actually needed. And then the non-API code would do the work of handling any appropriate conversions with the outside world of files and filenames, etc.

Reply:
> My pie-in-the-sky dream on that front would be that "API" functions in our codebase should manipulate actual Python objects, rather than, for example, strings that can be parsed/manipulated into the data that's actually needed. And then the non-API code would do the work of handling any appropriate conversions with the outside world of files and filenames, etc.

That's _exactly_ what past SCT devs (myself and Alexandru Jora) had in mind when we tried to delineate between [API and CLI code](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-API-code-vs-CLI-code). :smile: :heart: 

For example, https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/2869 outlines exactly this dream.

See also https://github.com/spinalcordtoolbox/spinalcordtoolbox/projects/7.